spec/controllers/generic_files_controller_spec.rb

```diff
@@ -21,14 +21,6 @@
         allow(GenericFile).to receive(:new).and_return(mock)
       end
 
-      it "should record on_behalf_of" do
```

Comment:
Yes.  The whole on_behalf_of got moved to the works level.  This will eventually need to be done for the generic file again.  Making transfers work for both generic work and generic file will be a bit more complicated and was out of scope for the ticket.